### PR TITLE
initial Amazon Connect event

### PR DESCRIPTION
**Pull request by msailes**
*Issue #, if available:*
https://github.com/aws/aws-lambda-java-libs/issues/106

*Description of changes:*
Initial event to represent integration with Amazon Connect.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ConnectEvent.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ * http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package com.amazonaws.services.lambda.runtime.events;
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
```
Comment (andreac30) [inline aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ConnectEvent.java:16]: As the other events I think `@NoArgsConstructor` is superfluous here.

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ConnectEvent.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+public class ConnectEvent implements Serializable, Cloneable {
+
+    private Details details;
+    private String name;
+
+    @NoArgsConstructor
+    @Data
+    public static class Details implements Serializable, Cloneable {
+        private ContactData contactData;
+        private Parameters parameters;
```
Comment (andreac30) [inline aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ConnectEvent.java:39]: Looking at the [url above](https://docs.aws.amazon.com/connect/latest/adminguide/connect-lambda-functions.html) regarding the `parameters` field:
- These are parameters specific to this call that were defined when you created the Lambda function.
In fact, looking at this [other example](https://aws.amazon.com/blogs/contact-center/creating-data-driven-ivrs-in-amazon-connect-using-a-rest-api/) it seems that the `parameters` field should be a `Map<String, Object>`. What do you think?

Comment (msailes) [reply aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ConnectEvent.java]: fixed

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ConnectEvent.java (hunk 1 above)
Comment (msailes) [reply aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ConnectEvent.java]: fixed

**Comment by msailes** [comment]
I tested this integration with Amazon Connect today.

Giving the following object.

```
2020-08-13T10:16:22.714+01:00
ConnectEvent(details=ConnectEvent.Details(contactData=ConnectEvent.ContactData(attributes=
{}
, channel=VOICE, contactId=6ec3e74d-8b43-46b4-a903-ba34d63274c7, customerEndpoint=ConnectEvent.CustomerEndpoint(address=+447763633581, type=TELEPHONE_NUMBER), initialContactId=6ec3e74d-8b43-46b4-a903-ba34d63274c7, initiationMethod=INBOUND, instanceArn=arn:aws:connect:eu-west-2:xxxxxxxxxxxxx:instance/eff7a619-3316-4273-8981-0c56a370c52d, previousContactId=6ec3e74d-8b43-46b4-a903-ba34d63274c7, queue=null, systemEndpoint=ConnectEvent.SystemEndpoint(address=+441202618222, type=TELEPHONE_NUMBER)), parameters=
{}
), name=ContactFlowEvent)
```

**Pull request merged**